metals/src/main/scala/scala/meta/internal/metals/BatchedFunction.scala

```diff
@@ -69,7 +85,7 @@ final class BatchedFunction[A, B](
     }
   }
   private def runAcquire(): Unit = {
-    if (lock.compareAndSet(false, true)) {
+    if (lock.compareAndSet(false, true) && !paused.get()) {
```

Comment:
I think we need to reverse the conditions to avoid locking the queue forever

```scala
     if (!paused.get() && lock.compareAndSet(false, true)) {
```

Reply:
Done